councilmatic_core/management/commands/convert_rtf.py

```diff
@@ -0,0 +1,114 @@
+import os
+import sys
+import subprocess
+import logging
+import logging.config
+import sqlalchemy as sa
+import datetime
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.db.models import Max
+
+from councilmatic_core.models import Bill
+
+logging.config.dictConfig(settings.LOGGING)
+logger = logging.getLogger(__name__)
+
+DB_CONN = 'postgresql://{USER}:{PASSWORD}@{HOST}:{PORT}/{NAME}'
+
+engine = sa.create_engine(DB_CONN.format(**settings.DATABASES['default']),
+                          convert_unicode=True,
+                          server_side_cursors=True)
+
+class Command(BaseCommand):
+    help = 'Converts rtf-formatted legislative text to valid html'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--update_all',
+            default=False,
+            help='Update full_text in all bills: can be set to True.')
+
+    def handle(self, *args, **options):
+        self.connection = engine.connect()
+        self.update_all = options['update_all']
+        '''
+        This command converts RTF from Legistar into valid HTML.
+        The conversion employs "unoconv" - a CLI tool that imports and exports documents in LibreOffice. We run unoconv as a daemon process and kill it when the conversions finish.
+        Three steps occur: (1) querying the database for bill full_text (i.e., the RTF from Legistar), 
+        (2) iteration over the query results, conversion to html, and creation of an inserts string,
+        (3) updating the full_text field with new html.
+        '''
+        listener = subprocess.Popen(['unoconv', '--listener'], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        try:
+            self.add_html()
+        finally:
+            listener.kill()
+
+    def get_rtf(self):
+        self.connection.execute("SET local timezone to '{}'".format(settings.TIME_ZONE))
+        with engine.begin() as connection:
+            # Only apply this query to most recently updated (or created) bills.
+            max_updated = Bill.objects.all().aggregate(Max('ocd_updated_at'))['ocd_updated_at__max']
+            # if max_updated is None or options['update_all']:
+            if max_updated is None or self.update_all:
+                max_updated = datetime.datetime(1900, 1, 1)    
```

Comment:
It would be a safer to have two versions of the query, one without the updated at in the where clause and one with.